### PR TITLE
Open-source optimizations

### PR DESCRIPTION
**Pull request by nir0s**
@wolf-daniel :)

Please ignore code coverage failure for now. Need to deepen coverage anyway.

### CODE REVIEW

**Comment by nir0s** [comment]
@wolf-daniel ping

**Comment by nir0s** [comment]
@wolf-daniel ping

File: .circleci/config.yml [hunk 1]
```diff
@@ -52,6 +53,13 @@ jobs:
     environment:
       - TOX_ENV: py34
 
+  test-2.7:
```
Comment (wolf-daniel) [inline .circleci/config.yml:57]: How does that resonate with running `python3` in `test-template`?

File: wryte.py [hunk 2]
```diff
@@ -66,23 +68,22 @@
 
 
 class JsonFormatter(logging.Formatter):
-    def __init__(self, pretty=False):
+    def __init__(self, pretty=False): # pylint: disable=super-init-not-called
```
Comment (wolf-daniel) [inline wryte.py]: Would we rather just disable this in pylint?

**Comment by wolf-daniel** [comment]
I see we set those 75% marks on coverage, but we seem to have recently stopped reaching them.
Shall we update the marks? Not sure how arbitrary these 75% marks really are.
Or do you want to address why we don't hit them and how we could?

**Comment by wolf-daniel** [comment]
Btw, personally, I wouldn't bother with managing coverage right now.
Since it already exists, it's great, but I feel we can be loose here.
We haven't really formalized our open source policy, so I don't think we should be too strict with this right now.

File: .circleci/config.yml (hunk 1 above)
Comment (nir0s) [reply .circleci/config.yml:57]: Well, I override the docker image and TOX env.

File: wryte.py (hunk 2 above)
Comment (nir0s) [reply wryte.py]: True.

**Comment by nir0s** [comment]
I'm used to having high coverage in my open source projects, but you're right, we should have a policy.

I'll set it to 50% as an arbitrary number that isn't 0 for the codebase and 0% for changes. It will keep us on our toes not to neglect this for too long.

File: .circleci/config.yml (hunk 1 above)
Comment (wolf-daniel) [reply .circleci/config.yml:57]: Yeah, but is `python3` command you run in the job template (as opposed to just running `python`) necessarily set to python 2 in the Circle image? The `TOX_ENV` isn't really relevant for this, I think.

**Pull request merged**